state/logs.go

```diff
@@ -546,33 +546,28 @@ func (t *logTailer) processReversed(query *mgo.Query) error {
 		return errors.Errorf("too many lines requested (%d) maximum is %d",
 			t.params.InitialLines, maxInitialLines)
 	}
+
 	iter := query.Sort("-t", "-_id").
 		Limit(t.params.InitialLines).
 		Iter()
-	defer iter.Close()
```

Comment:
We still need to close the iterator

Reply:
Thanks for pointing it out ian, missed the err could be returned preemptively in t.tomb.Dying():